Implement build --label by adding a LABEL instruction

tiborvass opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/docker/docker/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**\- What I did**

**\- How I did it**

**\- How to verify it**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Tibor Vass tibor@docker.com

icecrime commented:
```
15:27:29 # github.com/docker/docker/builder/dockerfile
15:27:29 builder/dockerfile/builder.go:227: err declared and not used
```

yongtang committed:
Labels set on the command line always override labels in Dockerfile

This fix tries to address the inconsistency in #22036 where labels
set on the command line will not override labels specified in
Dockerfile, but will override labels inherited from `FROM` images.

The fix add a LABEL with command line options at the end of the
processed Dockerfile so that command line options labels always
override the LABEL in Dockerfiles (or through `FROM`).

An integration test has been added for test cases specified in #22036.

This fix fixes #22036.

NOTE: Some changes are from #22266 (@tiborvass).

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>
(cherry picked from commit 5844736c14b29860ea03b040e9a052e59ad75bfc)

vdemeester commented:
LGTM 🐰

tiborvass commented:
Actually we should probably just add #22184 to #22242. cc @mlaventure
